Code/autopkglib/__init__.py

```diff
@@ -277,8 +293,10 @@ def set_pref(self, key, value):
         # On macOS, write it back to preferences domain if we didn't use a file
         if is_mac() and self.type is None:
             self._set_macos_pref(key, value)
-        elif is_windows() and self.file_path is not None:
+        elif self.file_path is not None:
             self.write_file()
+        else:
+            log(f"WARNING: Preference change {key}=''{value}'' was not saved.")
```

Comment:
Should this be log_err() instead?

Reply:
I'm not sure. It could be. This case will happen whenever you're running autopkg on Linux or Windows, the default config (such as `/home/username/.config/Autopkg/config.json`) doesn't exist and you haven't used the `--prefs` option. I don't believe it's possible for it to happen on mac, since either the preference domain will be used, or `--prefs` will have been passed to autopkg.

Thoughts?

Reply:
Even if this is not something that should happen commonly, this still suggests to me that it's an error condition. It even has "WARNING" in the text right there. If it's an error condition of any kind, I kinda feel like like it should be `log_err` and go to stderr.

I'm not super strongly focused on this, but we should be consistent. If there's enough precedence for other warnings going to stdout and not stderr, I'll drop it.

Reply:
Changed this to `log_err`. I agree about the `stdout` / `stderr` division.